components/common/src/ui.rs

```diff
@@ -43,6 +44,169 @@ pub const NONINTERACTIVE_ENVVAR: &str = "HAB_NONINTERACTIVE";
+        let msg = match *self {
+            UISymbolStyle::Full => "full",
+            UISymbolStyle::Limited => "limited",
+            UISymbolStyle::Ascii => "ascii",
+        };
+        write!(f, "{}", msg)
+    }
+}
+
+#[derive(Clone, Copy)]
+pub enum UISymbol {
+    UpArrow,
+    FingerPoint,
+    CheckMark,
+    BoxedCheckMark,
+    Omega,
+    BoxedX,
+    RightArrow,
+    Cloud,
+    DownArrow,
+    Elipses,
+    DottedTriangle,
+    RightShift,
+    Star,
+    SlashedZero,
+    ErrorX,
+}
+
+impl UISymbol {
+    pub fn to_str(&self) -> &str {
+        let style = if let Ok(s) = env::var(SYMBOL_STYLE_ENVVAR) {
+            match UISymbolStyle::from_str(&s) {
+                Ok(style) => style,
+                Err(e) => {
+                    eprintln!("{}\nSetting UISymbolStyle to {}",
+                              e,
+                              UISymbolStyle::default());
+                    UISymbolStyle::default()
+                }
+            }
+        } else {
+            UISymbolStyle::default()
+        };
+
+        match style {
+            UISymbolStyle::Ascii => {
+                match *self {
+                    UISymbol::UpArrow => "/^\\",
+                    UISymbol::FingerPoint => "-->",
+                    UISymbol::CheckMark => "[x]",
+                    UISymbol::BoxedCheckMark => "[#]",
+                    UISymbol::Omega => "-->",
+                    UISymbol::BoxedX => "-X-",
+                    UISymbol::RightArrow => "-->",
+                    UISymbol::Cloud => "-->",
+                    UISymbol::DownArrow => "  >",
+                    UISymbol::Elipses => "...",
+                    UISymbol::DottedTriangle => "???",
+                    UISymbol::RightShift => " >>",
+                    UISymbol::Star => "***",
+                    UISymbol::SlashedZero => "  0",
+                    UISymbol::ErrorX => "XXX",
+                }
+            }
+            UISymbolStyle::Limited => {
+                match *self {
+                    UISymbol::UpArrow => "↑",
+                    UISymbol::FingerPoint => "→",
+                    UISymbol::CheckMark => "√",
+                    UISymbol::BoxedCheckMark => "⌂",
+                    UISymbol::Omega => "Ω",
+                    UISymbol::BoxedX => "░",
+                    UISymbol::RightArrow => "→",
+                    UISymbol::Cloud => "⌂",
+                    UISymbol::DownArrow => "↓",
+                    UISymbol::Elipses => "…",
+                    UISymbol::DottedTriangle => "‼",
+                    UISymbol::RightShift => "»",
+                    UISymbol::Star => "≡",
+                    UISymbol::SlashedZero => "Ø",
+                    UISymbol::ErrorX => "XXX",
+                }
+            }
+            UISymbolStyle::Full => {
+                match *self {
+                    UISymbol::UpArrow => "↑",
+                    UISymbol::FingerPoint => "☛",
```

Comment:
Based on the way this is *used*, I think a better name would be `::InProgress`. Since the rendered glyphs are going to vary, the name should express the common concept, right?

Reply:
I had an initial draft where instead of using a literal term describing the glyph, I used a term describing what it is intended to indicate (like "progress"). What I personally found was that several of the glyphs could be called "progress" and in the end it just seemed simpler to call the symbol what it was for now and maybe move to a different approach later that trims down our glyphs to a smaller set with labels that indicates what action they represent.

Reply:
Again, I'm fine with moving this work to a follow-up PR if it's going to happen immediately after this. I think we we have a number of different glyphs that all mean "progress", we should consider whether using different ones improves the user experience, or if it would be more illustrative to consolidate them.